### PR TITLE
added support for EU region (endpoint option)

### PR DESCRIPTION
**Pull request by mlison**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Context

Mailgun uses a separate API endpoint for EU domains. Trying to authenticate with an EU sending API key against the US endpoint does not work (error 401).

### Changes

Add step option to chose your endpoint. Defaults to US (api.mailgun.net).

### CODE REVIEW

**Review by Bence1001 — COMMENTED** [review]
Hi there! 👋 Thank you for your contribution!

I've added some minor comments, but apart from that it LGTM!

I'm going to ask our tech writer team to review the title and the description.

File: step.yml [hunk 1]
```diff
@@ -44,6 +44,15 @@ inputs:
       summary: ""
       is_required: true
       is_sensitive: true
+  - api_endpoint: "api.mailgun.net"
+    opts:
+      title: "Mailgun API Endpoint"
+      description: |-
+        Defaults to US (api.mailgun.net)
```
Comment (zoltan-baba) [inline step.yml:51]:
```suggestion
        The API endpoint to be used with Mailgun: US or EU. 

		The default value is the US endpoint (api.mailgun.net).
```

**Comment by zoltan-baba** [comment]
Hi @mlison! I've added a suggestion for the description to be a bit more verbose.

**Comment by Bence1001** [comment]
```diff
diff --git a/bitrise.yml b/bitrise.yml
--- bitrise.yml
+++ bitrise.yml
@@ -103,9 +103,9 @@
             # Check for unhandled errors
             go get -u -v github.com/kisielk/errcheck
 
             # Go lint
-            go get -u -v github.com/golang/lint/golint
+            go get -u -v golang.org/x/lint/golint
 
   # ----------------------------------------------------------------
   # --- Utility workflows
   dep-update:

```

**Comment by Bence1001** [comment]
@mlison The CI is currently failing because it's trying to use an old version of golint. Could you please update the one line I've commented above ☝️ 

I'm also looking into why the CI's status is not reported back to GitHub. Once that's fixed we can merge the PR! 😊

**Review by Bence1001 — APPROVED** [review]
LGTM! Thank you! 🙏

**Pull request merged**